### DISCUSSION TITLE
set_security: Enable io.RawIOBase as an alternative to a string-based path to the certificates

### DISCUSSION THREAD

**Discussion by parallaxe** [FreeOpcUa/opcua-asyncio#1114] [Ideas]
In cases where the certificates are only in-memory or not explicitly stored inside a file, it is a bit cumbersome to create a temporary file just to call set_security.
My suggestion is to add `io.RawIOBase` as another alternative for `certificate`, `private_key` and `server_certificate`.

Any opinions on that? If you think this is worthwhile, I would start implementing it as soon as possible.

**Comment by oroulet** [comment]
No problem
